### PR TITLE
Add UI test and comment for end of lesson logic

### PR DESCRIPTION
**Pull request by maureensturgeon**
Testing and comments added for review https://github.com/code-dot-org/code-dot-org/pull/44506#pullrequestreview-863029788

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/EndOfLessonDialog.jsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ function EndOfLessonDialog({lessonNumber, isSummaryView}) {
       handleClose={handleClose}
       style={styles.dialog}
     >
-      <h2 id="uitest-end-of-lesson-header">
+      <h2 className="uitest-end-of-lesson-header">
```
Comment (jamescodeorg) [inline apps/src/templates/EndOfLessonDialog.jsx:32]: Just curious, why did you change this from an id to a class?

**Review by jamescodeorg — APPROVED** [review]
Looks great, thanks for adding these!

File: apps/src/templates/EndOfLessonDialog.jsx (hunk 1 above)
Comment (maureensturgeon) [reply apps/src/templates/EndOfLessonDialog.jsx:32]: I noticed in other places we were using classes for the uitest tag
